mpi-proxy-split/lower-half/mmap64.c

```diff
@@ -80,13 +114,49 @@ getMmappedList(int *num)
   return mmaps;
 }
 
+// When the parent started this child process, it then called
+//   split_process.cpp:startProxy() to write the lh_memRange
+//   start and end values.  We already read it here within
+//   first_constructor() in lh_proxy.
+// Then the parent copied the child process into the parent
+//   at split_process.cpp:read_lh_proxy_bits(pid_t().
+// Now we live in the parent process, and the upper half is calling
+//   split_process.cpp:initializeLowerHalf(), which is calling
+//   resetMmappedList() in this file.
 void
 resetMmappedList()
 {
+  // Alternatively, just do:  memset((void *)mmaps, 0, sizeof(mmaps));
   for (int i = 0; i < numRegions; i++) {
     memset(&mmaps[i], 0, sizeof(mmaps[i]));
   }
   numRegions = 0;
+
+  // We will reserve the lh_memRange memory, so no one can steal it from us.
+  size_t length = lh_memRange.end - lh_memRange.start;
+#if HAS_MAP_FIXED_NOREPLACE
+  lh_memRange.start = mmap(lh_memRange.start, length,
+             PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_FIXED_NOREPLACE, -1, 0);
+  if (lh_memRange.start == MAP_FAILED) {
+    char msg[] = "*** Panic: MANA lower half: can't reserve lh_memRange"
+                 " using MAP_FIXED_NOREPLACE\n";
+    perror("mmap");
+    write(2, msg, sizeof(msg)); assert(rc == 0);
```

Comment:
I think rc is not defined in this context? @gc00

Reply:
Can you fix it directly now and test?
I'll update the PR next.
- Gene

Reply:
I commented it out and am testing the rest of the code, will send email once I gather everything I observed. @gc00